TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -61,6 +63,36 @@ public override bool BeginCodeElement(CodeElement codeElement)
             return false;
         }
 
+        public override bool Visit(End end)
+        {
+            // detect CodeElement with a mix of Debug and "Normal" lines in debugging mode         
+            if (DebuggingMode && end.CodeElement.Type != CodeElementType.SentenceEnd)
+            {
+                CodeElement openingCodeElement = end.Parent.CodeElement;
+                if (openingCodeElement != null)
+                {
+                    CodeElement endCodeElement = end.CodeElement;
+                    bool isStartDebug = IsCodeElementInDebug(openingCodeElement);
+                    bool isEndDebug = IsCodeElementInDebug(endCodeElement);
+                    if (isStartDebug == isEndDebug && openingCodeElement is CodeElements.IfStatement)
```

Comment:
This limit the check only on IfStatements, Is it what we want to do in this issue ?